kvm/kvm.go

```diff
@@ -246,11 +261,131 @@ type Device struct {
 }
 
 // CreateDev creates an emulated device in the kernel.
-// The file descriptor returned in fd can be used with KVM_SET/GET/HAS_DEVICE_ATTR.
+// The file descriptor returned in fd can be used with SET/GET/HAS_DEVICE_ATTR.
 func CreateDev(vmFd uintptr, dev *Device) error {
 	_, err := Ioctl(vmFd,
 		IIOWR(kvmCreateDev, unsafe.Sizeof(Device{})),
 		uintptr(unsafe.Pointer(dev)))
 
 	return err
 }
+
+// Translation is a struct for TRANSLATE queries.
+type Translation struct {
+	// LinearAddress is input.
+	// Most people call this a "virtual address"
+	// Intel has their own name.
+	LinearAddress uint64
+
+	// This is output
+	PhysicalAddress uint64
+	Valid           uint8
+	Writeable       uint8
+	Usermode        uint8
+	_               [5]uint8
+}
+
+// TranslateAddr translates a virtual address according to the vcpu’s current address translation mode.
```

Comment:
This refactoring by moving from machine to kvm package makes sense. Thanks.

nit:
```suggestion
// Translate translates a virtual address according to the vcpu’s current address translation mode.
```

Reply:
done